v7/_posts/odata features/2017-12-26-06-00-JsonBatch.md

```diff
@@ -0,0 +1,330 @@
+    
+Objects required for writing response:
+
+{% highlight csharp %}
+
+    odataResponseMessage : IODataResponseMessage    //backed by the underlying HTTP response stream.
+    odataMessageWriterSettings: ODataMessageWriterSettings
+    
+    ODataMessageWriter batchResponseMessageWriter = new ODataMessageWriter(odataResponseMessage, odataMessageWriterSettings, model);
+    ODataBatchWriter batchWriter = batchResponseMessageWriter.CreateODataBatchWriter();
+{% endhighlight %}
+    
+Similar to the request message writer, the response message writer above also sets up the proper `ODataPayloadKind` for Batch processing.
+
+Now, start the basic batch processing with request reading / response writing. For batch processing, the request processing (including reading, dispatching) and response writing (including processing responses for individual request, writing batch response) are interleaving and the whole process is driven by OData service using specific call patterns as described below (common for both formats).
+
+Typically odataBatchReader is started with initial state, so we can start with creating the response envelop, followed by readings & writings.
+
+{% highlight csharp %}
+    
+        batchWriter.WriteStartBatch();
+
+        while (batchReader.Read())
+        {
+            switch (batchReader.State)
+            {
+                case ODataBatchReaderState.Operation:
+
+                    // Create operation request message
+                    ODataBatchOperationRequestMessage operationRequestMessage = batchReader.CreateOperationRequestMessage();
+
+                    // Create operation response message
+                    IODataResponseMessage responseMessage = batchWriter.CreateOperationResponseMessage(operationRequestMessage.currentContentId);    
+
+                    // Request processing begins: including request body reader creation, request parsing & execution according to dependencies specified etc. belonged
+                    // to implementation of specific services.
+                    if (operationMessage.Method == "PUT")
+                    {
+                        // Processing request payload.
+                        using (ODataMessageReader operationMessageReader = new ODataMessageReader(
+                            operationMessage, new ODataMessageReaderSettings(), this.userModel))
+                        {
+                            ODataReader reader = operationMessageReader.CreateODataResourceReader();
+								
+                        // Reader reads data from request payload and kick off service backend processing.
+                            ...
+                        }
+
+                        // Query various attributes for this operation request such as request dependencies and atomic group associated
+                        // and combine with specific processing of the service.
+                        Foo_ProcessingForExecutionOrders(operationMessage.DependsOnIds);
```

Comment:
I don't think we're supposed to use "Foo".  Maybe make it "MyProcessingForExecutionOrders(...)?

Reply:
no foobar for sure, but I think foo is OK now. Here I think My... is better so I have updated it.